Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -398,6 +399,36 @@ - (void)writeAvatarToDisk:(UIImage *)avatar
     });
 }
 
+- (NSData *)processedImageDataForRawAvatar:(UIImage *)image
+{
+    NSUInteger kMaxAvatarBytes = 5 * 1000 * 1000;
+
+    if (image.size.width != kOWSProfileManager_MaxAvatarWidth
```

Comment:
That's a good point. 

A couple thoughts:

1. We should handle group avatars the same way since it's fundamentally the same problem.
2. We still want to maintain a 1:1 AR. So e.g. if you upload a 100x200px avatar, the uploaded avatar will be 100x100 max.